contracts/contracts/TACoApplication.sol

```diff
@@ -0,0 +1,708 @@
+                _operator == _stakingProvider || getBeneficiary(_operator) == address(0),
+                "Specified operator is a provider"
+            );
+            // Set new operator->stakingProvider relation
+            _stakingProviderFromOperator[_operator] = _stakingProvider;
+        }
+
+        if (info.operatorStartTimestamp == 0) {
+            stakingProviders.push(_stakingProvider);
+        }
+
+        if (info.operatorConfirmed) {
+            authorizedOverall -= info.authorized;
+        }
+
+        // Bond new operator (or unbond if _operator == address(0))
+        info.operator = _operator;
+        info.operatorStartTimestamp = uint64(block.timestamp);
+        emit OperatorBonded(_stakingProvider, _operator, previousOperator, block.timestamp);
+        _releaseOperator(_stakingProvider);
+    }
+
+    /**
+    * @notice Make a confirmation by operator
+    */
+    function confirmOperatorAddress() external {
+        address stakingProvider = _stakingProviderFromOperator[msg.sender];
+        require(isAuthorized(stakingProvider), "No stake associated with the operator");
+        StakingProviderInfo storage info = stakingProviderInfo[stakingProvider];
+        require(!info.operatorConfirmed, "Operator address is already confirmed");
+        require(msg.sender == tx.origin, "Only operator with real address can make a confirmation");
+
+        updateRewardInternal(stakingProvider);
+        info.operatorConfirmed = true;
+        authorizedOverall += info.authorized;
+        emit OperatorConfirmed(stakingProvider, msg.sender);
+
+        if (address(updatableStakeInfo) != address(0)) {
+            updatableStakeInfo.updateOperator(stakingProvider, msg.sender);
+        }
+    }
+    
+    //-------------------------XChain-------------------------
+
+    /**
+    * @notice Resets operator confirmation
+    */
+    function _releaseOperator(address _stakingProvider) internal {
+        stakingProviderInfo[_stakingProvider].operatorConfirmed = false;
+        if (address(updatableStakeInfo) != address(0)) {
+            updatableStakeInfo.updateOperator(_stakingProvider, address(0));
+        }
+    }
+
+    /**
+    * @notice Send updated authorized amount to xchain contract
+    */
+    function _updateAuthorization(address _stakingProvider, StakingProviderInfo storage _info) internal {
+        if (address(updatableStakeInfo) != address(0)) {
+            // TODO send both authorized and eligible amounts in case of slashing from StakeInfo
+            uint96 eligibleAmount = getEligibleAmount(_info);
+            updatableStakeInfo.updateAmount(_stakingProvider, eligibleAmount);
+        }
+    }
+
+    //-------------------------Slashing-------------------------
+    /**
+    * @notice Slash the provider's stake and reward the investigator
+    * @param _stakingProvider Staking provider address
+    * @param _penalty Penalty
+    * @param _investigator Investigator
+    */
+    function slash(
```

Comment:
Let's discuss (not for this PR) if we should not launch with slashing (or alternatively, deactivate slashing by tuning the parameters). See https://github.com/nucypher/nucypher-contracts/issues/104